### PR TITLE
Fix the init command

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose

Fixes:https://github.com/ballerina-platform/ballerina-library/issues/6195

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingInitTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingInitTest.java:2]:
```suggestion
 *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
```

File: persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java [hunk 2]
```diff
@@ -85,55 +79,20 @@ public void execute() {
...
-                    ". but found '%s' datasource.%n", Arrays.toString(SUPPORTED_DB_PROVIDERS.toArray()), datastore);
-            return;
-        }
-
         if (Files.isDirectory(Paths.get(sourcePath, PERSIST_DIRECTORY, MIGRATIONS))) {
             errStream.println("ERROR: reinitializing persistence after executing the migrate command is not " +
                     "permitted. please remove the migrations directory within the persist directory and try " +
                     "executing the command again.");
             return;
         }
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java:87]: Shall we remove this too? As we are not adding any configuration, this validation is not valid

Comment (Bhashinee) [reply persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java]: Fixed in [5a7297f](https://github.com/ballerina-platform/persist-tools/pull/336/commits/5a7297f297e0a3e36d42d91d70bb4bbf78856fb1)

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingInitTest.java (hunk 1 above)
Comment (Bhashinee) [reply persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingInitTest.java]: Fixed in [5a7297f](https://github.com/ballerina-platform/persist-tools/pull/336/commits/5a7297f297e0a3e36d42d91d70bb4bbf78856fb1)

File: persist-cli/src/main/resources/cli-help/ballerina-persist-generate.help [hunk 3]
```diff
@@ -14,5 +14,5 @@ EXAMPLES
           $ bal persist generate --help
 
        Generate client objects for the model definition file inside persist directory.
-          $ bal persist generate
+          $ bal persist generate --module persist --datastore mysql
```
Comment (daneshk) [inline persist-cli/src/main/resources/cli-help/ballerina-persist-generate.help:17]: SYNOPSIS
                bal persist generate [--module <module name>]
                                        [--datastore <store type>]
                                        [<-h> | <--help>]

OPTIONS
       --module <module name>
              The name of the module to initialize the persistence layer. If not specified, the default module will be used.
       --datastore <store type>
              The type of the datastore to be used for persistence.
              The supported datastores are 'inmemory', 'mysql', 'mssql', 'postgresql', 'googlesheets', and redis.
       -h, --help
           Print the usage details of all commands.

Comment (Bhashinee) [reply persist-cli/src/main/resources/cli-help/ballerina-persist-generate.help:28]: Fixed in 48cc647

File: persist-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 4]
```diff
@@ -44,7 +58,9 @@ OPTIONS
 BALLERINA COMMANDS
        The below is a list of available subcommands:
 
-       init            Initialize the package for persistence
+       add             Initialize the package for persistence. Used along with `bal build` command.
```
Comment (daneshk) [inline persist-cli/src/main/resources/cli-help/ballerina-persist.help:61]:
```suggestion
       add             Initialize the package for persistence and integrate the client generation to the `bal build` command.
```

Comment (Bhashinee) [reply persist-cli/src/main/resources/cli-help/ballerina-persist.help]: Addressed the suggestions in [103a302](https://github.com/ballerina-platform/persist-tools/pull/336/commits/103a3022b5e3836c115a576dec082b7d2f2116c4)

**Review by daneshk — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**